### DISCUSSION TITLE
Add field export link/button on model Import pages?

### DISCUSSION THREAD

**Discussion by cmcknz77** [netbox-community/netbox#16197] [Ideas]
One of the things that I've always found frustrating about NetBox is the lack of a 'light-effort' way to get hold of the column headers on any of the data Import pages. (I'm currently still on 3.7.7).

So, say I want to import a whole bunch of Device Roles. The import page for that particular model gives me the Field Options table shown here (minus one custom field).
![image (7)](https://github.com/netbox-community/netbox/assets/8886905/5640ecd9-2d2c-4b69-895d-e21bcea7e5bd)

But because the field names are all in column 1, I have to copy the entire table from the page, drop it into something like Excel and then transpose it and then delete the data that I didn't need or vice-versa, just in order to get the column headers that I need to use.
I know I read [somewhere](https://github.com/netbox-community/netbox/wiki/Frequently-Asked-Questions#why-cant-i-pass-exported-data-directly-into-the-import-form) that there's no current method to export the live data from any particular model that includes and uses the correct import field names instead of the 'humanized' display names.

I can think of 2 ways to make this easier.
1. In the Field Options section but above, or below the table, print a comma-separated list of just the correctly formatted field names. In the pictured case (for Device Roles) it'd just read: name,slug,color,vm_role,config_template,description,tags,id,cf_addcustomfield
or
2. Add a button to each import page allowing the downloading of a correctly formatted but 'blank' .csv (or .json or .yaml) file (respecting the Format and CSV delimiter dropdowns) containing just the correctly formatted field names (so in this case, assuming that .csv is chosen format, it'd generate a text file with the name 'import.csv' and the contents followed by a new line would be: name,slug,color,vm_role,config_template,description,tags,id,cf_addcustomfield
Unless I'm not seeing the wood for the trees in which case can someone please point out where I'm making things unnecessarily hard for myself?

Pro for Option1 - It'd be real easy
Con for Option1 - Difficult to make look pretty especially for models with large numbers of fields
Pro for Option2 - Minimizes cluttering the gorgeous new v4.x.x UI
Con for Option2 - I don't know if it's possible

**Comment by KMTMark** [comment]
Was just looking at this myself - I'm currently doing a bit of local testing for a plugin we're considering using but wanted to see how it would look with our own data so thought I'd export the min. I can get away with from our Prod versions and import to my local Docker one - but have come up against this weird issue as Netbox export gives the CSV & YAML headers with the uppercase formatting but the import expects them all lowercase

**Reply by candlerb** [reply]
> have come up against this weird issue as Netbox export gives the CSV & YAML headers with the uppercase formatting but the import expects them all lowercase

I agree it's weird, but it's an intentional design decision by the Netbox authors that the export and import formats are different and mutually incompatible. It's documented in the [FAQ](https://github.com/netbox-community/netbox/wiki/Frequently-Asked-Questions#why-cant-i-pass-exported-data-directly-into-the-import-form).

**Comment by KMTMark** [comment]
Ah thanks!
In my case atm I'm just doing some local testing via docker so was hoping
to be able to import data from our current setup.

On Tue, 21 May 2024 at 17:01, Brian Candler ***@***.***>
wrote:

> have come up against this weird issue as Netbox export gives the CSV &
> YAML headers with the uppercase formatting but the import expects them all
> lowercase
>
> I agree it's weird, but it's an intentional design decision by the Netbox
> authors that the export and import formats are different and mutually
> incompatible. It's documented in the FAQ
> <https://github.com/netbox-community/netbox/wiki/Frequently-Asked-Questions#why-cant-i-pass-exported-data-directly-into-the-import-form>
> .
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/netbox-community/netbox/discussions/16197#discussioncomment-9510584>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APHEWNQJEYL6HDJFVSAQVFDZDNVXHAVCNFSM6AAAAABH6NZJD2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TKMJQGU4DI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
Many thanks,


*Mark Skrzypczyk *(He/Him)
Desktop Support Engineer

kheironmed.com  <http://kheironmed.com/>| Transforming cancer diagnostics
through the power of deep learning


Tel: +44 20 7039 3500 #923 | Timezone: London (GMT/BST)

Winner of the UK Government's AI in Health and Care Award 2020
<https://www.kheironmed.com/news/https/www.nhsx.nhs.uk/news/nhs-ai-lab-speed-cancer-and-heart-care/>

-- 










Kheiron Medical Technologies Ltd. is a registered company in 
England and Wales. This e-mail and its attachment(s) are intended for the 
above named only and are confidential. If they have come to you in error 
then you must take no action based upon them but contact us immediately. 
Any disclosure, copying, distribution or any action taken or omitted to be 
taken in reliance on it is prohibited and may be unlawful. Although this 
e-mail and its attachments are believed to be free of any virus, it is the 
responsibility of the recipient to ensure that they are virus free. If you 
contact us by e-mail then we will store your name and address to facilitate 
communications. Any statements contained herein are those of the individual 
and not the organisation.




Registered number: 10184103. Registered 
office: WeWork, The Bower, 207 Old Street, London, England, EC1V 9NR

**Reply by candlerb** [reply]
You can't transfer data between Netbox instances using CSV. There are lots of reasons why, including some circular references (e.g. `device <- interface <- IP address <- device`), the inability to name things uniquely, and the inability to select the IDs of items you are importing.

Instead, you need to dump the postgres database from one instance, and import it into another. This process is documented under the heading [Replicating Netbox](https://docs.netbox.dev/en/stable/administration/replicating-netbox/) in the Administration section of the documentation.
